lib/puppet/pal/catalog_compiler.rb

```diff
@@ -37,6 +35,14 @@ def with_json_encoding(pretty: true, exclude_virtual: true)
       yield JsonCatalogEncoder.new(catalog, pretty: pretty, exclude_virtual: exclude_virtual)
     end
 
+    # Calls a block of code and yields a data hash representing the compiled catalog
+    # to the block.
+    #
+    # @api public
+    def catalog_data_hash
+      catalog.to_data_hash
+    end
```

Comment:
The above impl is fine - the `to_data_hash` returns a copy - it does not have to yield to a code block.
The comment above is however not true to the implementation. Either change so it does what the comment says, or change the comment ;-)
A unit test would be nice as well.